presto-hive/src/test/java/com/facebook/presto/hive/util/TestHiveFileIterator.java

```diff
@@ -0,0 +1,230 @@
+        hadoopConf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+
+        listDirectoryOperation = path ->
+        {
+            FileSystem fs = path.getFileSystem(hadoopConf);
+            return new HadoopFileInfoIterator(fs.listLocatedStatus(path));
+        };
+    }
+
+    private void createFiles(String basePath, int numFiles, boolean randomName, String fileNamePrefix) throws IOException
+    {
+        for (int i = 0; i < numFiles; i++) {
+            new File(basePath).mkdirs();
+            String fileName;
+            if (randomName) {
+                fileName = getRandomStringOfLen(10);
+            }
+            else {
+                fileName = fileNamePrefix + "_" + i;
+            }
+            new File(basePath + File.separator + fileName).createNewFile();
+        }
+    }
+
+    private List<File> createDirs(String basePath, int numDirectories)
+    {
+        List<File> directories = new ArrayList<>();
+        for (int i = 0; i < numDirectories; i++) {
+            String dirName = basePath + File.separator + pathFilterString + "_" + getRandomStringOfLen(5);
+            File file = new File(dirName);
+            file.mkdirs();
+            directories.add(file);
+        }
+        return directories;
+    }
+
+    private String getRandomStringOfLen(int len)
+    {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < len; i++) {
+            sb.append(RANDOM_FILE_NAME_SALT_STRING.charAt(random.nextInt(RANDOM_FILE_NAME_SALT_STRING.length())));
+        }
+        return sb.toString();
+    }
+
+    private void updateExpectedStats(HiveFileIterator.NestedDirectoryPolicy policy, boolean hasPathFilter, int expectedCount)
+    {
+        switch (policy) {
+            case IGNORED:
+                if (hasPathFilter) {
+                    fileCountForIgnoredPolicyPathFilterOn += expectedCount;
+                }
+                else {
+                    fileCountForIgnoredPolicyPathFilterOff += expectedCount;
+                }
+                break;
+            case RECURSE:
+                if (hasPathFilter) {
+                    fileCountForRecursePolicyPathFilterOn += expectedCount;
+                }
+                else {
+                    fileCountForRecursePolicyPathFilterOff += expectedCount;
+                }
+                break;
+        }
+    }
+
+    public int getFileCount(HiveFileIterator hiveFileIterator)
+    {
+        int actualCount = 0;
+        while (hiveFileIterator.hasNext()) {
+            hiveFileIterator.next();
+            actualCount++;
+        }
+        return actualCount;
+    }
+
+    @Test
+    public void testNoPathFilterNoRecursion()
+    {
+        hiveFileIterator = new HiveFileIterator(rootPath, listDirectoryOperation, namenodeStats, IGNORED, Optional.empty());
+        int actualCount = getFileCount(hiveFileIterator);
+        assertEquals(actualCount, fileCountForIgnoredPolicyPathFilterOff);
```

Comment:
> By introducing something like `hive.input-path-filter-class` we are explicitly encouraging jar drop-ins. This is something that we were historically trying to avoid (however it is still possible,e.g.: with storage formats). The "presto way" would be to extend the `presto-hive` module, and have an interface in `presto-hive` that allows to extend this behaviour.
> 
> @bhasudha What is your usecase? Do you use `presto-hive` connector as is? Or do you have some wrapper on top of it?

@arhimondr I understand. We are using `presto-hive-hadoop2` with drop-ins for custom storage format. The pathfilter class in this case will also be coming from Hudi jars- [HoodieROTablePathFilter](https://github.com/apache/incubator-hudi/blob/master/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieROTablePathFilter.java). 

Like @shixuan-fan suggested let me try to introduce a PathFilterProvider interface.

Reply:
@bhasudha Did you consider having a `presto-hive-hoodie` connector wrapper in your proprietary codebase? This way you can verify jar dependency compatibility during the build time. Also you can inject whatever extensions you need without introducing runtime class resolutions that can be very error prone.

Reply:
@arhimondr Wrapper connector is a good idea. I can try that. Just wanted to clarify that this is not proprietary. Presto users in Hudi open source community also face this issue today. Since we have already solved this at Uber by patching Presto, thought generalizing that would be useful for everyone. 
I can refactor this change into a separate module called 'presto-hudi' connector and update the PR. Let me know if you are okay with that.

Reply:
Sounds good